### DISCUSSION TITLE
is PINECONE_API_KEY required? / suggestions for comments in configuration files

### DISCUSSION THREAD

**Discussion by ClaireCJS** [TransformerOptimus/SuperAGI#245] [General]
Does it just use local storage otherwise?

Can we put a comment into the config.yaml template answering this question so people won't have the question in the first place when editing the template? :)  

I'm guessing the answer is no because there is a waitlist for new users currently which would mean most people couldn't jump into this project, but it would be nice for a comment to be there because I'm scared it is required and that i should just give up until the waitlist :)

And I'm guessing the answer is no because STORAGE_TYPE below: is set to "FILE"

* It would be nice if this file was more self-documenting, so everything people need to know about the file is in the file while editing file. Perhaps these answers are elsewhere?

- is YOUR_EMAIL_APP_PASSWORD required?
- 
- What is the pro/con of using GITHUB_OATH?  
- What is the pro/con of using S3 vs FILE storage? {I can guess, but someone else probably wants to know}

- What is the advantage of getting  a GOOGLE_API_KEY? I have one, but the file says "IF" you have an api key and CSE key, use this, which makes it sound not required.  And I don't know what a CSE key is.

- I can't figure out what a CSE key is, would be nice if the comment mentioned that, maybe a link to the site to grab one (like how EMAIL_PASSWORD has a very helpful link to myaccount.google.com/apppasswords)

- Could use an example of what YOUR_DRAFTS_FOLDER should be set to, though I think it's intuitive. What might help would be to have some sample drafts in a sample folder and have the project already point to an existing (generic) set that we could the modify without having to figure out as much. [Basically, chamber our round so we can shoot our gun, right now it's feeling like having to learn how to work a bullet press, but that's just how I'm personally experiencing it because I'm more of a beginner with github.]
- 
- In the original setup instructions it says: "Google API key	Create a project in the Google Cloud Console and enable the API you need (for example: Google Custom Search JSON API)." Could we just add a 
sentence or two about how we would know which APIs we need, or what some examples or, so that we can be inspired to have ideas in which direction to take this?  I'm just installing this for curiosity, so I don't know which API i may or may not need.

I think this would help more beginner users launch the project with more confidence and get farther in a happier and more-informed state.
-

**Comment by neelayan7** [comment]
Hey. Thanks for raising this. We have moved all our configurations for the tools in the GUI along with extensive Readme's for beginners to use SuperAGI effectively. Do let us know if you have any other questions.

**Reply by chris-aeviator** [reply]
could you please elaborate on the OP question if pinecone is required?

**Comment by adamphetamine** [comment]
This question needs a direct answer- your documentation says

'3. Enter your LLM API Key or Auth token in config.yaml file.'

I've added an OpenAI key into the .yaml file but it is not being recognised.

Looking for a solution I find this- 
'We have moved all our configurations for the tools in the GUI '

So which is it?

**Reply by JohnHunt999** [reply]
You can add your OpenAI key in the settings in the GUI and it should work.

**Reply by adamphetamine** [reply]
Thanks for the quick reply @JohnHunt999 
Unfortunately this still doesn't answer the question

- if we CAN add the OpenAI key to the .yaml file, how can I troubleshoot the fact it's not working?
- if we MUST add the OpenAI key via the GUI, why do the instructions still ask us to modify the .yaml?

**Reply by JohnHunt999** [reply]
In the last 2 releases we added the option to use open sourced models after which there might have been some issue with reading the api key from yaml, which shouldn't have happened will look into it. And also we should have updated the readme.

**Reply by adamphetamine** [reply]
Thanks again.
I've had this issue last 2 releases, if I figure it out I'll add the answer here.
If the recommended method is to get the project running and add these tokens later, I'm happy with that but would be super happy if it was explicit in the docs.

Thanks again
